src/import-export/import-match-map.c

```diff
@@ -306,6 +373,7 @@ Account* gnc_imap_find_account_bayes(GncImportMatchMap *imap, GList *tokens)
     struct account_info account_i;
     kvp_value* value;
     kvp_frame* token_frame;
+    gboolean guid_bayes = gnc_features_check_used (imap->book, GNC_FEATURE_GUID_BAYESIAN);
```

Comment:
I'm a little worried about keying off of GNC_FEATURE_GUID_BAYESIAN instead of using a new feature because GUID_BAYESIAN has been in place since 2.6.12, meaning that someone could open a 2.8.x book with a GnuCash version between that and 2.6.18, in which case GnuCash wouldn't be able to read the Bayesian import map and would write data to it in the old format.

Reply:
Is gnucash actually taking initiative to start *using* guid's for bayesian in 2.6.12 or is it only able to handle them if it gets to open a file that is already using guid's for baysian ? The latter would be safe, the former would indeed be a problem.

Reply:
IIUC Bob's approach was to be able to read guid accounts in 2.6.12. Any writes would be with names and would unset the feature flag so that the next time the book was opened in 2.7/8 it would go through and fix all of the "new" entries.

The new code in PR 233 changes the format of the KVP keys in a way that I think 2.6.12-2.6.18 won't understand but since those versions know about the feature flag will open the book anyway and the import matcher will generate "non-flat" tokens from the import and they might not match the flattened tokens, so that transactions that should match, won't. It would also unset the feature flag and the next time the book is opened with GnuCash 2.7/8 the PR233 conversion code will be presented with a mixture of flat and not-flat token keys representing what should be the same key.

Reply:
You're right. So what we should do is  indeed use another feature key instead that supercedes GNC_FEATURE_GUID_BAYESIAN and will only be known starting from whenever this gets merged.
That will mean the GNC_FEATURE_GUID_BAYESIAN key will never be used in practice (except for a few people that have been experimenting with early 2.7 releases. So in theory we can drop GNC_FEATURE_GUID_BAYESIAN as no stable release code will ever set it.

Reply:
Bob's plan was to have gnucash 2.6 be able to read both styles, but write in the "old" fashion creating a mixed KVP. This pull request (in concern with the other pull request) makes it so that only 2.7 and up can convert an import map to GUID, and 2.6 can read to and write to it in the appropriate style.

@jralls, I see what you're saying. A file created in 2.7 then read in 2.6.18 will not be interpreted properly. As @gjanssens says, a new feature flag that supersedes the old flag makes sense. I'll do some more thinking about it.

Reply:
I added a short design document here: https://wiki.gnucash.org/wiki/User:Lmat It has a question in it; would you please weigh in there?

Reply:
I have added my input to your short design document.